lib/db/sstable_builder.cc

```diff
@@ -0,0 +1,91 @@
+#include "db/sstable_builder.hh"
+#include <cstddef>
+#include <cstdint>
+#include <cstring>
+#include <iostream>
+#include <map>
+#include <memory>
+#include <ostream>
+#include <string>
+#include <type_traits>
+#include <unistd.h>
+
+using namespace dibibase::db;
+
+uint8_t SSTableBuilder::m_total_sstables = 0;
+
+SSTableBuilder::SSTableBuilder(std::multimap<std::string, std::string> memtable)
+    : m_memtable(memtable) {
+  ++m_total_sstables;
+  char *buffer = new char[100];
+  encode_data(buffer);
+  m_files =
+      std::unique_ptr<SSTableFiles>(new SSTableFiles(buffer, m_total_sstables));
+  delete[] buffer;
+}
+
+void SSTableBuilder::encode_data(char *buffer) {
+  // data format:
+  // first byte represents nu. of entires in memtable.
+  // for each entry: [key length, key characters, value length, value
+  // characters].
+
+  std::string mem_size = std::to_string(m_memtable.size());
```

Comment:
**Suggestion**
we can always store the size in a fixed number of bytes 4 or 8, this should solve the problem of multidigit size 
```suggestion
  std::string mem_size = std::bitset<64>(m_memtable.size()).to_string();
```
also, we can convert it back in the same way
```
uint64_t size = std::bitset<64>(mem_size).to_ulong();
```